components/airlock/src/util.rs

```diff
@@ -98,7 +99,7 @@ pub fn check_user_group_membership(username: &str) -> Result<()> {
     for grp in vec!["tty"].iter() {
         let user_group =
             users::get_group_by_name(grp).ok_or(Error::GroupNotFound(String::from(*grp)))?;
-        if !user_group.members().contains(&user) {
+        if !user_group.members().contains(&OsString::from(&user)) {
```

Comment:
I think you can eliminate the need to construct a new `user` string and use just the `&str` parameter with the implicit conversion:
```suggestion
        if !user_group.members().contains(&username.into()) {
```
And with this, you may not need the `use std::ffi::OsString;` either.

Reply:
That doesn't work. I'm not sure that I completely understand how the borrower works in rust yet, but I think that consumes the value so it's then not available to be used down on line 103.

Reply:
Huh, I'm not sure without looking deeper what the issue is, but if it's not a simple change, the way you have it should work.